packages/requests/src/lib/requests-result.ts

```diff
@@ -88,18 +107,15 @@ function updateRequestResult(key: unknown[], newValue: Partial<Result>) {
 
     for (const key of Object.keys(newValue)) {
       if (
-        currentResult[key as keyof Result] !== newValue[key as keyof Result]
+        currentResult[key as keyof RequestResult] !==
+        newValue[key as keyof RequestResult]
       ) {
         hasChange = true;
         break;
       }
     }
 
-    hasChange &&
-      result.next({
-        ...currentResult,
-        ...newValue,
```

Comment:
We can use a spread but we need to make 4 options to be required to update. We can't update just like that ` updateRequestResult(key, { isLoading: true });` because if there was `isSuccess: true` before then it'll be a wrong state.

Reply:
We can leave it for now.

The last feature is staleTime. Do you want to do it? We can maintain the lasUpdateDate on the result and compare.

Reply:
I have no time today and be able only on Monday.

Reply:
No problem